Public/Src/Tools/UnitTests/Analyzers/XLGToDBAnalyzerTests.cs

```diff
@@ -69,5 +74,62 @@ public void DBCreatedAndPopulated()
 
             XAssert.AreNotEqual(Directory.GetFiles(OutputDirPath.ToString(Context.PathTable), "*.sst").Length, 0);
         }
+
+        /// <summary>
+        /// This test makes sure that the number of events logged (per event type) match 
+        /// what is expected based on the build schedule that we manually populate and build.
+        /// </summary>
+        [Fact]
+        public void DBEventCountVerification()
+        {
+            Configuration.Logging.LogExecution = true;
+
+            var fileOne = FileArtifact.CreateOutputFile(Combine(TestDirPath, "foo.txt"));
+            var fileTwo = FileArtifact.CreateOutputFile(Combine(TestDirPath, "bar.txt"));
+            var dirOne = CreateOutputDirectoryArtifact(Combine(TestDirPath, "baz").ToString(Context.PathTable));
+            Directory.CreateDirectory(ArtifactToString(dirOne));
+
+            var pipA = CreateAndSchedulePipBuilder(new Operation[]
+            {
+                Operation.WriteFile(fileOne),
+            }).Process;
+
+            var pipB = CreateAndSchedulePipBuilder(new Operation[]
+            {
+                Operation.WriteFile(fileTwo),
+            }).Process;
+
+            var pipC = CreateAndSchedulePipBuilder(new Operation[]
+            {
+                Operation.ReadFile(fileOne),
+                Operation.WriteFile(CreateOutputFileArtifact())
+            }).Process;
+
+            var pipD = CreateAndSchedulePipBuilder(new Operation[]
+            {
+                Operation.EnumerateDir(dirOne),
+                Operation.WriteFile(CreateOutputFileArtifact())
+            }).Process;
+
+            var buildA = RunScheduler().AssertSuccess();
+
+            var analyzerRes = RunAnalyzer(buildA).AssertSuccess();
+
+            var dataStore = new XldbDataStore(storeDirectory: OutputDirPath.ToString(Context.PathTable));
+
+            XAssert.AreEqual(dataStore.GetFileArtifactContentDecidedEvents().Count(), 5);
+            XAssert.AreEqual(dataStore.GetWorkerListEvents().Count(), 0);
+            XAssert.AreEqual(dataStore.GetPipExecutionPerformanceEvents().Count(), 4);
+            XAssert.AreEqual(dataStore.GetDirectoryMembershipHashedEvents().Count(), 0);

```

Comment:
>XAssert.AreEqual(dataStore.GetDirectoryMembershipHashedEvents().Count(), 0); [](start = 12, length = 76)

I'm surprised this event isn't logged since pipD enumerates a directory. This should cause BuildXL to need to compute a directory fingerprint when storing the pip to cache after it executes

Reply:
yeah, this is puzzling... @Sahil can you add some dummy source (existing prior build starts) files under that directory? Maybe there is some weird optimization is going on, and we report nothing for an empty dir?

---
In reply to: [310238006](https://github.com/microsoft/BuildXL/pull/694#discussion_r310238006) [](ancestors = 310238006)

Reply:
Other thoughts (no need to investigate/address in this PR) - I bet `Operation.EnumerateDir` calls into some .Net API for enumeration, and we saw in the past that in some really rare cases Detours had troubles with catching the accesses (I remember switching some operation(s) from calling into .net to invoking winapi directly)

---
In reply to: [310242591](https://github.com/microsoft/BuildXL/pull/694#discussion_r310242591) [](ancestors = 310242591,310238006)